src/main/markdown.ts

```diff
@@ -0,0 +1,128 @@
+import { getGlobalRouteRegistry, RouteDefinition } from './router';
+import { groupBy } from './util';
+
+export function generateMarkdownApiSpec() {
+    const doc: string[] = [];
+    const routes = getGlobalRouteRegistry().filter(_ => !_.isMiddleware);
+    const groupByPath = groupBy(routes, r => r.path);
+    for (const [_path, routes] of groupByPath) {
+        for (const route of routes) {
+            doc.push(...generateEndpointSpec(route));
+        }
+    }
+    return doc;
+}
+
+function generateEndpointSpec(endpoint: RouteDefinition): string[] {
+    const method = endpoint.method.toUpperCase();
+    let header = `## ${method} ${endpoint.path}`;
+    if (endpoint.deprecated) {
+        header += ' **(deprecated)**';
+    }
+
+    const { queryParams, bodyParams } = generateParamSpec(endpoint);
+    const responses = generateResponsesSpec(endpoint);
+
+    const spec: string [] = [
+        '\n',
+        header,
+        endpoint.summary.trim(),
+    ];
+
+    if (queryParams.length > 0) {
+        spec.push('\n', '### Query Params', ...queryParams);
+    }
+
+    if (bodyParams.length > 0) {
+        spec.push('\n', '### Body Params', ...bodyParams);
+    }
+
+    if (responses.length > 0) {
+        spec.push('\n', '### Responses', ...responses);
+    }
+
+    return spec;
+}
+
+function generateParamSpec(endpoint: RouteDefinition) {
+    const bodyParams: string[] = [];
+    const queryParams: string[] = [];
+
+    for (const p of endpoint.params) {
+        let header = '- ' + p.name;
+
+        if (p.required) {
+            header += ' (*required*)';
+        }
+
+        if (p.deprecated) {
+            header += ' (*deprecated*)';
```

Comment:
Yeah, this one should be bold, and probably come even before the param name, hehe